admin/src/scenes/classe/header/VerifClassButton.tsx

```diff
@@ -41,24 +41,23 @@ export default function VerifClassButton({ classe, setClasse, isLoading, setLoad
   };
 
   const verifyClasse = async () => {
-    try {
-      setLoading(true);
-
-      const { ok, code, data } = await api.put(`/cle/classe/${classe?._id}/verify`, classe);
-
-      if (!ok) {
-        toastr.error("Oups, une erreur est survenue lors de la vérification de la classe", translate(code));
-        return setLoading(false);
-      } else {
+    setLoading(true);
+    await apiv2
+      .post<ClassesRoutes["GetOne"]["response"]["data"]>(`/classe/${classe?._id}/verify`, classe)
+      .then((updatedClasse) => {
+        console.log("updatedClasse", updatedClasse);
         toastr.success("Opération réussie", "La classe a bien été vérifiée");
-      }
-      setClasse(data);
-    } catch (e) {
-      capture(e);
-      toastr.error("Oups, une erreur est survenue lors de la vérification de la classe", e);
-    } finally {
-      setLoading(false);
-    }
+        const classeView = ClasseService.mapDtoToView(updatedClasse);
+        setClasse(classeView);
+      })
+      .catch((error: HttpError) => {
+        if (error?.statusCode === 422) {
```

Comment:
```suggestion
        if (error instanceof BusinessError && error.statusCode === 422) {
        } else if (error instanceof TypeError) {
        }
```
plutot se base sur les types ?